core/modules/locale/tests/locale.test

```diff
@@ -2808,3 +2808,46 @@ class LocaleLanguageNegotiationInfoFunctionalTest extends BackdropWebTestCase {
     $this->backdropGet('admin/config/regional/language/detection');
   }
 }
+
+/**
+ * Tests locale translation safe string handling.
+ */
+class LocaleStringIsSafeTest extends DrupalWebTestCase {
```

Comment:
Good catch. I am also not an expert coder, but what you say make sense to me. I'll see if I can change it to extend `BackdropWebTestCase` and see what happens with and without compatibility enabled.